src/karts/controller/skidding_ai.cpp

```diff
@@ -1172,80 +1182,228 @@ void SkiddingAI::handleItems(const float dt)
-            if( m_time_since_last_shot < 3.0f) break;
+    // Angle to aim_point
+    Vec3 kart_aim_direction = *aim_point - m_kart->getXYZ();
 
-            // Use bubblegum if the next kart  behind is 'close' but not too close
-            // (too close likely means that the kart is not behind but more to the
-            // side of this kart and so won't be hit by the bubble gum anyway).
-            // Should we check the speed of the kart as well? I.e. only drop if
-            // the kart behind is faster? Otoh this approach helps preventing an
-            // overtaken kart to overtake us again.
-            if(m_distance_behind < 15.0f && m_distance_behind > 3.0f    )
-            {
-                m_controls->setFire(true);
-                m_controls->setLookBack(true);
-                break;
-            }
+    // Make sure we have a valid last_node
+    if(last_node==Graph::UNKNOWN_SECTOR)
+        last_node = m_next_node_index[m_track_node];
 
-            // If this kart is in its last lap, drop bubble gums at every
-            // opportunity, since this kart won't envounter them anymore.
-            LinearWorld *lin_world = dynamic_cast<LinearWorld*>(World::getWorld());
-            if(m_time_since_last_shot > 3.0f &&
-                lin_world &&
-                lin_world->getKartLaps(m_kart->getWorldKartId())
-                                   == race_manager->getNumLaps()-1)
-            {
-                m_controls->setFire(true);
-                m_controls->setLookBack(true);
-                break;
-            }
-            break;   // POWERUP_BUBBLEGUM
-        }
+    int node = m_track_node;
+    float distance = 0;
+    std::vector<const Item *> items_to_collect;
+    std::vector<const Item *> items_to_avoid;
+
+    // 1) Filter and sort all items close by
+    // -------------------------------------
+    const float max_item_lookahead_distance = 10.f;
+    while(distance < max_item_lookahead_distance)
+    {
+        int n_index= DriveGraph::get()->getNode(node)->getIndex();
+        const std::vector<Item *> &items_ahead =
+            ItemManager::get()->getItemsInQuads(n_index);
+        for(unsigned int i=0; i<items_ahead.size(); i++)
+        {
+            evaluateItems(items_ahead[i],  kart_aim_direction,
+                          &items_to_avoid, &items_to_collect);
+        }   // for i<items_ahead;
+        distance += DriveGraph::get()->getDistanceToNext(node,
+                                                      m_successor_index[node]);
+        node = m_next_node_index[node];
+        // Stop when we have reached the last quad
+        if(node==last_node) break;
+    }   // while (distance < max_item_lookahead_distance)
+
+    //items_to_avoid and items_to_collect now contain the closest item information needed after
+    //What matters is (a) if the lists are void ; (b) if they are not, what kind of item it is
+   
+    switch( m_kart->getPowerup()->getType() )
```

Comment:
Since this code has grown a lot, it may make sense to split the handling of each powerup type in its own method to avoid this method growing too big

Reply:
Indeed, separate methods for each powerup could be a good idea. Especially as I expect it to grow even more.

Reply:
My previous comment about splitting the by case-when into sub-methods still holds